jetty-websocket/jetty-websocket-client/src/main/java/org/eclipse/jetty/websocket/client/WebSocketClient.java

```diff
@@ -239,6 +239,7 @@ public long getMaxTextMessageSize()
     public void setIdleTimeout(Duration duration)
     {
         configurationCustomizer.setIdleTimeout(duration);
+        getHttpClient().setIdleTimeout(duration.toMillis());
```

Comment:
@sbordet how about this?
The javadoc for this is `The duration that a websocket may be idle before being closed by the implementation`, but this is really before the websocket even opens.

Or should I leave it so anyone wanting to set this will need to do 
```java
client.getHttpClient().setIdleTimeout(1000);
```

Reply:
@lachlan-roberts if `WebSocketClient` has a `setIdleTimeout(Duration)` method, then it's likely a new one introduced in Jetty 10, and yes it should be forwarded to `HttpClient` to cover for the case for upgrade request sent, but never coming back.

However, I can imagine people wanting to set different idle timeouts for the HTTP communication and for WebSocket communication, so `wsClient.getHttpClient().setIdleTimeout(long)` would allow that.